app/controllers/users_controller.rb

```diff
@@ -654,15 +662,18 @@ def set_user
   end
 
   def user_scope
-    if helpers.moderator?
+    if current_user&.at_least_moderator?
       User.all
     else
       User.active
     end.joins(:community_user).includes(:community_user, :avatar_attachment)
   end
 
   def check_deleted
-    if (@user.deleted? || @user.community_user.deleted?) && (!helpers.moderator? || params[:deleted_screen].present?)
+    deleted = @user.deleted? || @user.community_user.deleted?
+    go_to_not_found = !current_user&.at_least_moderator? || params[:deleted_screen].present?
```

Comment:
Untested suggestion:

```suggestion
    go_to_not_found = current_user&.standard? || params[:deleted_screen].present?
```